Grbl_Esp32/Custom/CoreXY.cpp

```diff
@@ -90,9 +90,25 @@ bool user_defined_homing(uint8_t cycle_mask) {
     pl_data->motion.systemMotion   = 1;
     pl_data->motion.noFeedOverride = 1;
 
-    for (int cycle = 0; cycle < 3; cycle++) {
-        AxisMask mask = homing_cycle[cycle]->get();
+    uint8_t cycle_count = (cycle_mask == 0) ? 3 : 1;  // if we have a cycle_mask, we are only going to do one axis
+
+    AxisMask mask = 0;
+    for (int cycle = 0; cycle < cycle_count; cycle++) {
+        // if we have a cycle_mask, do that. Otherwise get the cycle from the settings
+        if (cycle_mask == 0) {
+            mask = homing_cycle[cycle]->get();
+        } else {
+            mask = cycle_mask;
+        }
+
+        // If not X or Y do a normal home
+        if (!(mask & bit(X_AXIS) || mask & bit(Y_AXIS))) {
```

Comment:
Please use `!(bitnum_istrue(mask, X_AXIS) || bitnum_istrue(mask, Y_AXIS)` .  I went to a fair amount of effort to regularize all the bit operators around the bit_ and bitnum_ macros; I hate to see the grovel-y versions creeping back in.